### PR TITLE
Fix edge case where column name is same as alias name

### LINKED ISSUE #130: Column name is same as Alias name in LCA
Below query shouldn't be warned for invalid LCA, since table also has the same column name.

``` python
select ca_zip
     from (
      SELECT substr(ca_zip,1,5) ca_zip
      FROM customer_address
      WHERE substr(ca_zip,1,5) IN ('89436', '30868')

### PR DESCRIPTION
**Pull request by bishwajit-db**
Fix edge case where column name is same as alias name. Fixes #130

### CODE REVIEW

File: src/databricks/labs/remorph/snow/dialect_utils.py [hunk 1]
```diff
@@ -60,26 +51,43 @@ def check_for_unsupported_lca(
...
+        if window_expr:
+            window_expressions.append(window_expr)
+    return window_expressions
+
+
+def _find_aliases_in_select(select_expr: Select) -> dict[str, bool]:
+    aliases = {}  # Alias name and if it is same as a column name used in the alias expression
+    for expr in select_expr.expressions:
+        if isinstance(expr, exp.Alias):
+            aliases[expr.output_name] = any(column.name == expr.output_name for column in expr.find_all(exp.Column))
```
Comment (nfx) [inline src/databricks/labs/remorph/snow/dialect_utils.py:67]: Unfold any to a simpler code

File: tests/unit/snow/test_dialect_utils.py [hunk 2]
```diff
@@ -119,3 +119,18 @@ def test_query_with_error():
...
     error = check_for_unsupported_lca(dialect, sql, filename)
     assert not error
+
+
+def test_query_with_same_alias_and_column_name():
+    dialect = "SNOWFLAKE"
+    sql = """
+    select ca_zip
+     from (
+      SELECT substr(ca_zip,1,5) ca_zip
```
Comment (nfx) [inline tests/unit/snow/test_dialect_utils.py:129]: Add another test case where you have "expr AS alias"

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/snow/dialect_utils.py (hunk 1 above)
Comment (bishwajit-db) [reply src/databricks/labs/remorph/snow/dialect_utils.py]: Converted to a for loop

File: tests/unit/snow/test_dialect_utils.py [hunk 3]
```diff
@@ -119,3 +119,18 @@ def test_query_with_error():
 
     error = check_for_unsupported_lca(dialect, sql, filename)
     assert not error
+
+
+def test_query_with_same_alias_and_column_name():
+    dialect = "SNOWFLAKE"
+    sql = """
+    select ca_zip
```
Comment (sundarshankar89) [inline tests/unit/snow/test_dialect_utils.py:127]: can you add test case where there are more than one column as well as for window expression

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: tests/unit/snow/test_dialect_utils.py (hunk 3 above)
Comment (bishwajit-db) [reply tests/unit/snow/test_dialect_utils.py:127]: Added the window expression in test

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Comment by sundarshankar89** [comment]
@nfx ready for review

**Review by nfx — APPROVED** [review]

**Pull request merged**